internal/generate/olm-catalog/descriptor/descriptor.go

```diff
@@ -119,16 +150,57 @@ func GetCRDDescriptionForGVK(apisDir string, gvk schema.GroupVersionKind) (olmap
 	return crdDesc, nil
 }
 
-// getTypesFromDir gets all Go types from dir.
-func getTypesFromDir(dir string) (types.Universe, error) {
+func isDirExist(path string) (bool, error) {
```

Comment:
The semantics of this function are weird, since it returns an error if the dir exists but has some stat error. Its callers should be expected to handle the case of the dir existing but not being accessible.

Reply:
> Its callers should be expected to handle the case of the dir existing but not being accessible

@estroz That's exactly the only case when an error is returned here i.e stat error when file exists but inaccessible (permission or filesystem) error.
```Go
if err != nil {
	if os.IsNotExist(err) {
		return false, nil
	}
	return false, err
}
```
So not sure what you wanted to change about the error return for this function.

All the callers of this function are currently "handling" this "exists-but-inaccessible" error by throwing it all the way to the top causing `generate csv` to exit with an error. 
Although on double checking, one caller in `GetCRDDescriptionForGVK()` was ignoring this error but I've fixed that.

Reply:
Given the name, I would expect something like:

```go
func isDirExist(path string) bool {
	info, err := os.Stat(path)
	if err != nil {
		return os.IsExist(err)
	}
	return info.IsDir()
}
```

Considering how you're using this function, a better name would be `isDirAccessible`.